src/block-management/ethrpc-transport-adapter.js

```diff
@@ -0,0 +1,41 @@
+"use strict";
+
+var nextToken = 1;
```

Comment:
This is not the same token that is inside `AbstractTransport`.  You will find that this pattern is followed a lot in the various "classes" that were created in this PR, the previous one, and the upcoming one.  The tokens are not interchangeable in any way and each stands on its own.  In all cases, they are used to allow the caller to unsubscribe from something he previously subscribed to.  Reconnects, blocks, logs, etc.  If a function like `foo.subscribe` returns a token, that token is only useful for calling `foo.unsubscribe`.  It is not useful for calling `bar.unsubscribe` or `zip.removeReconnectListener`.

> could subscriptionMapping be replaced with an array

This would lead to a sparse array over time if someone is connecting/reconnecting which isn't good for JavaScript performance (though pragmatically it probably wouldn't matter much at the moment).

> since the tokens are just integers

The returned tokens are explicitly strings not numbers.  The fact that they happen to be auto-incremented numbers is an implementation detail and subject to change.  A UUI with class prefix would probably be a better system for generating tokens as it would allow the unsubscribe methods to validate that the token provided was reasonable, but I didn't feel writing the code associated with or bringing in the dependencies required to make that happen.